libc/test/UnitTest/LibcTest.cpp

```diff
@@ -38,7 +38,8 @@ TestLogger &operator<<(TestLogger &logger, Location Loc) {
 // When the value is UInt128, __uint128_t or wider, show its hexadecimal
 // digits.
 template <typename T>
-cpp::enable_if_t<cpp::is_integral_v<T> && (sizeof(T) > sizeof(uint64_t)),
+cpp::enable_if_t<(cpp::is_integral_v<T> && (sizeof(T) > sizeof(uint64_t))) ||
+                     cpp::is_big_int_v<T>,
```

Comment:
What's the difference between the LHS of the `||` and RHS? Does the LHS not hold for the RHS?

Reply:
Quoting [cpp_reference for std::is_integral](https://en.cppreference.com/w/cpp/types/is_integral#:~:text=Provides%20the%20member,behavior%20is%20undefined.):

> _Provides the member constant value which is equal to true, if T is the type bool, char, char8_t(since C++20), char16_t, char32_t, wchar_t, short, int, long, long long, or any implementation-defined extended integer types, including any signed, unsigned, and cv-qualified variants. Otherwise, value is equal to false._
>
> _If the program adds specializations for std::is_integral or std::is_integral_v, the behavior is undefined._

This means that `BigInt` is `false` for `is_integral` (that's the purpose of this patch)
This also means that `__int128_t` / `__uint128_t` is `true`.

So `cpp::is_integral_v<T> && (sizeof(T) > sizeof(uint64_t)))` means `__int128_t` or `__uint128_t`.
And `cpp::is_big_int_v<T>` means `BigInt` instantiation.

Reply:
> And cpp::is_big_int_v<T> means BigInt instantiation.

Ah!

> (that's the purpose of this patch)

Ok, now I see how it all fits together.  Sorry for being slow on the uptake, but I do appreciate the patient explanations!

---

There's no equivalent to `is_big_int_v` in the `std::` namespace.  I was under the impression that our `cpp::` is meant to mirror things in `std::`?  As that was the intent behind these todos added in https://github.com/llvm/llvm-project/pull/84035/commits/48b0bc837085a38ff1de33010d9222363f70238f.  If those should be moved out of bit.h, should they (and `cpp::is_big_int<T>`) also be moved out of the `cpp::` namespace?

https://libc.llvm.org/dev/code_style.html#setting-errno-from-runtime-code was the only reference I found to any kind of `cpp::` convention; you would know better than I what our project's intent is with `cpp::`.

Or, perhaps am I conflating what does/does not belong under `libc/src/__support/CPP` with what does/does not belong in the `cpp::` namespace?

Reply:
> Ok, now I see how it all fits together. Sorry for being slow on the uptake, but I do appreciate the patient explanations!

No worries! It's important to speak up when something is unclear, and I'm happy to provide explanations or discuss rationale earlier than later.

> should they (and cpp::is_big_int<T>) also be moved out of the cpp:: namespace?

I think so, there is a bit of cleanup left. I'll take care of it. Things have grown organically for a while and we're slowly paying the technical debt.

Reply:
Ok, I'll work on finishing cleaning up the todos from https://github.com/llvm/llvm-project/commit/48b0bc837085a38ff1de33010d9222363f70238f next week.